### PR TITLE
Fix PocketMine-MP compatibility

### LINKED ISSUE #24: Incompatible with pmmp
As I remember, it worked with pmmp some time ago. Now I'm getting this error, because CID bans has been removed from pmmp (where they ever implemented?).

```
[18:38:06] [Server thread/CRITICAL]: Could not execute task synapsepm\TickSynapseTask: Call to undefined method pocketmine\Server::getCIDBans()
[18:38:06] [Server thread/CRITICAL]: Error: "Call to undefined method pocketmine\Server::getCIDBans()" (EXCEPTION) in "/SynapsePM_v1.0.3.phar/src/synapse/Player" at line 93
```

### PR DESCRIPTION
**Pull request by MichaelM04**
Fixes all compatibility issues with PocketMine-MP and undefined $spawnPosition error.

Fixes #24 

**_This has been fully tested and works perfectly_**

### CODE REVIEW

File: src/synapse/Player.php [hunk 1]
```diff
@@ -194,7 +194,9 @@ protected function processLogin(){
 			$pk->y = $this->y;
 			$pk->z = $this->z;
 			$pk->seed = -1;
-			$pk->dimension = $this->level->getDimension();
+			if($this->server->getName() != "PocketMine-MP"){
```
Comment (dktapps) [inline src/synapse/Player.php]: and what happens when it's a spoon of PocketMine-MP which doesn't have these methods?

**Review by dktapps — DISMISSED** [review]

File: src/synapse/Player.php (hunk 1 above)
Comment (MichaelM04) [reply src/synapse/Player.php]: Not many don't have these methods, in fact the only I can recall is steadfast, regardless, I believe I have a solution.

**Comment by NyaomiDEV** [comment]
Is it fully compatible? I had some class-related problems that I described in that issue.

**Comment by MichaelM04** [comment]
@AryToNeX I've tested using Player functions with no error, so it should work for you as well.

**Comment by tarik02** [comment]
Thanks.

**Comment by MichaelM04** [comment]
After looking through the changes from the pull a few minutes ago, I realised I had forgotten to fix the getCIDBans() with the new method, the commit is here. https://github.com/MichaelM04/SynapsePM/commit/e2866bab8e573e13c6a0fec9620a9d4f5bd6c0bc

**Pull request merged**